### PR TITLE
add authentication check middleware

### PR DESCRIPTION
**Pull request by stegzilla**

### CODE REVIEW

File: middleware/ensure_auth_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	handler.ServeHTTP(rw, req)
+
+	require.Equal(s.T(), rw.Code, 200)
+}
+
+func TestEnsureAuthTestSuite(t *testing.T){
+	suite.Run(t, new(EnsureAuthTestSuite))
+}
+
+func servePublicKey() jwk.Key {
```
Comment (danni-popova) [inline middleware/ensure_auth_test.go:97]: why is this here?

Comment (stegzilla) [reply middleware/ensure_auth_test.go:97]: had to wrap the tests in a suite so it only has to start the server once for all the tests, serves up the public key so token can be validated in test
